be/src/column/array_column.cpp

```diff
@@ -66,7 +66,10 @@ void ArrayColumn::resize(size_t n) {
 }
 
 void ArrayColumn::assign(size_t n, size_t idx) {
-    DCHECK(false) << "array column shouldn't call assign";
+    DCHECK_LE(idx, this->size()) << "Range error when assign arrayColumn.";
+    auto datum = get(idx);
+    this->reset_column();
```

Comment:
When this column is reset is the datum still valid? IMO datum is a reference not a deep copy obj

Reply:
fix that by copy the datum to a new column then swap.